src/site/content/en/web-dev-basics-one/js-libraries-vs-frameworks/index.md

```diff
@@ -0,0 +1,271 @@
+
+JavaScript library and framework usage is prolific across the web. Every other website seems to use some third-party code as part of its JavaScript resources. Web page [weight is getting worse](https://almanac.httparchive.org/en/2021/page-weight#fig-3) with time, which [affects users](https://timkadlec.com/remembers/2020-04-21-the-cost-of-javascript-frameworks/). [JavaScript](https://almanac.httparchive.org/en/2021/javascript) is a large contributing factor toward overall page weight, and it's this same JavaScript that often comprises third-party libraries and frameworks.
+
+It's not good enough to say, "Stop using JavaScript frameworks," because frameworks provide a big benefit to developers. Frameworks can help you to code efficiently and deliver features quickly, among other benefits. Instead, you should educate yourself so that you can make an informed decision when the time arises. 
+
+"Should I use a library or framework today?" is an uncommon question to ask yourself. Libraries and frameworks are two very different things. However, libraries and frameworks are often conflated and the more knowledge you have about the two, the more likely you are to make educated decisions about their use.
+
+## Examples of libraries and frameworks
+
+You might notice third-party code by other names, such as widgets, plugins, polyfills, or packages. However, they all typically fall into the category of a library or a framework. Basically, the difference between the two can be summarized as follows:
+
+-  For a library, your app code calls the library code.
+-  For a framework, [your app code is called by the framework](https://en.wikipedia.org/wiki/Inversion_of_control).
+
+### Library
+
+Libraries tend to be simpler than frameworks and offer a narrow scope of functionality. If you pass an input to a method and receive an output, you probably used a library.
+
+Look at this example of the [`lodash`](https://lodash.com/) library:
+
+```js
+import lodash from 'lodash'; // [1]
+const result = lodash.capitalize('hello'); // [2]
+console.log(result); // Hello
+```
+
+As is the case with many libraries, it's practical to read through this code and understand what it does. [There's very little magic involved](https://en.wikipedia.org/wiki/Magic_(programming)):
+
+1. An `import` statement imports the lodash library into the JavaScript program.
+1. The `capitalize()` method is invoked. 
+1. A single argument is passed to the method. 
+1. The return value is captured in a variable.
+
+### Framework
+
+Frameworks tend to be bigger than libraries and contribute more to overall page weight. In fact, a framework can include a library. 
+
+This example shows a plain framework without a library and uses [Vue](https://vuejs.org/), which is a [popular](https://2021.stateofjs.com/en-US/libraries/front-end-frameworks/#front_end_frameworks_experience_ranking) JavaScript framework:
+
+```html
+<!-- index.html -->
+<div id="main">
+  {{ message }}
```

Comment:
```suggestion
  {{ '{{ message }}' }}
```

Looks like the web.dev templating system, via eleventy, was interpolating the double mustache tags which broke the code sample. I've cloned this branch locally, have run the build, and can confirm we can escape it with this syntax `{{ '{{ message }}' }}` which fixes the issue. Screenshot confirmation of the rendered page below:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/381495/170238484-eb338496-0222-455a-939f-90c4a9802bfe.png">

Happy to use another technique though.